sympy/assumptions/handlers/matrices.py

```diff
@@ -38,6 +38,8 @@ def MatMul(expr, assumptions):
         factor, mmul = expr.as_coeff_mmul()
         if all(ask(Q.symmetric(arg), assumptions) for arg in mmul.args):
             return True
+        if ask(Q.diagonal(expr), assumptions):
```

Comment:
This is necessary for the expression `V1.T*(V1 + V2)` product of non symmetric matrices - dot product:

```
V1 = MatrixSymbol('V1', 2, 1)
V2 = MatrixSymbol('V2', 2, 1)
assert ask(Q.symmetric(V1.T*(V1 + V2))) is True
```